### ISSUE TITLE
let task output files outside its working directory

### ISSUE THREAD

**Issue by mlin** [chanzuckerberg/miniwdl#214]
Right now the task runtime doesn't allow output files that reside outside of the initial working directory or a subdirectory thereof; for example, attempting to output a file under /tmp will fail.

The constraint has to do with our approach of mounting a host directory as the read/write container working directory, and collecting the output files from the host directory after the container is auto-removed. Any files outside of the working directory get destroyed along with the container. We could revise this turn off container auto_remove, and then copy other output files out of the container file system before destroying the container ourselves. The downsides are (i) having to copy the files, and (ii) potentially neglecting to destroy the container under some code paths / circumstances.

**Comment by mlin** [comment]
Example to noodle on -- a straightforward `samtools index` or `tabix` task. The input files are mounted a bit outside of the working directory. When we run the indexing program, it generates the index file next to it, which we can't output.

A different solution is to mount input files in or under the working directory, like Cromwell. This could also be more convenient for having the index files in the "right place" by default, although that depends on not having name collisions. Also mild concern about uncontrollable potential for collision with files/directories the task command is hard-coded to create.

**Comment by mlin** [comment]
Possible heuristic for index files: following the current directory assortment pass, if (i) a file Idx has a basename which is equal to the basename of another file Dat plus one or more filename extensions; (ii) there exists no so-named Dat' file in the same directory as Idx; and (iii) there exists no file with the same basename as Idx in the directory of Dat; then, mount Idx in the same directory as Dat.

**Comment by vsmalladi** [comment]
This fails for the test in [relative_and_absolute_task.wdl](https://github.com/openwdl/wdl/blob/wdl-1.1/SPEC.md#files-and-optional-outputs)

```wdl
version 1.1

task relative_and_absolute {
  command <<<
  mkdir -p my/path/to
  printf "something" > my/path/to/something.txt
  >>>

  output {
    String something = read_string("my/path/to/something.txt")
    File bashrc = "/root/.bashrc"
  }

  runtime {
    container: "ubuntu:focal"
  }
}
```

Fails with error:
`error: "OutputError", message: "task outputs attempted to use a path outside its working directory: /root/.bashrc"`

**Comment by peterhuene** [comment]
FWIW, Sprocket will also fail on this in the same manner:

```
error: failed to evaluate output `bashrc` for task `relative_and_absolute`: guest path `/root/.bashrc` is not within a container mount
   ┌─ foo.wdl:11:10
   │
11 │     File bashrc = "/root/.bashrc"
   │          ^^^^^^
```

I don't think this is something that can be universally supported by all backends (namely TES, which currently requires that you know the paths to the output files _at task creation time_ which is not how WDL works; our support for TES will likely have to put the working directory as the output ahead of time).

**Comment by vsmalladi** [comment]
@peterhuene is it a output issue or more of mounting issue and path with root?

**Comment by mlin** [comment]
Yeah, this has always seemed to me *not close* to worthwhile in terms of the cost-benefit, namely complexity of implementation versus actual usefulness.
